src/mount.rs

```diff
@@ -0,0 +1,84 @@
+use request::Request;
+use response::Response;
+use middleware::{Continue, Middleware, MiddlewareResult};
+
+use hyper::uri::RequestUri::AbsolutePath;
+
+pub struct Mount {
+    mount_point: String,
+    mounted_handler: Box<Middleware + Send + Sync + 'static>,
+}
+
+fn rewrite_mount_point(mount_point: &str) -> String {
+    let mut mount_point = mount_point.to_owned();
+    if let Some('/') = mount_point.chars().last() {
+        let _ = mount_point.pop();
+    }
+    if let Some('/') = mount_point.chars().nth(0) {
+        mount_point
+    } else {
+        mount_point.insert(0, '/');
+        mount_point
+    }
+}
+
+impl Mount {
+    ///
+    /// Creates a new middleware that mounts a middleware at a mount point.
+    /// An incoming request that matches the mount point will be forwareded to
+    /// the mounted middleware, but with the path rewritten so that the mount
+    /// point appears to be the root from the perspective of the mounted
+    /// middleware. This can be useful in combination with the
+    /// `StaticFilesMiddleware`, for example.
+    ///
+    ///
+    /// # Examples
+    /// ```{rust}
+    /// use nickel::{Nickel, StaticFilesHandler, Mount};
+    /// let mut server = Nickel::new();
+    ///
+    /// server.utilize(
+    ///     Mount::new("/static_files/",
+    ///                StaticFilesHandler::new("/path/to/serve/")
```

Comment:
May be worth introducing a trait `Mountable` and implement it for `Nickel` so we can have `server.mount(foo, bar)`.